job/job.go

```diff
@@ -248,9 +248,41 @@ func (j *Job) Disable() {
 	j.Disabled = true
 }
 
+// DeleteFromParentJobs goes through and deletes the current job from any parent jobs.
+func (j *Job) DeleteFromParentJobs(cache JobCache) error {
+	j.lock.Lock()
+	defer j.lock.Unlock()
+	if len(j.ParentJobs) != 0 {
+		for _, p := range j.ParentJobs {
+			parentJob, err := cache.Get(p)
+			parentJob.lock.Lock()
+			defer parentJob.lock.Unlock()
+			if err != nil {
+				return err
+			}
+			ndx := 0
+			for i, id := range parentJob.DependentJobs {
+				if id == j.Id {
+					ndx = i
+					break
```

Comment:
@ajvb Great.  So when a Job `j` is deleted, you look at that Job's parents and delete references to `j` from said parents, and each parent only has 1 reference to `j` as you just mentioned.

What happens to jobs that `j` is a parent of?  Will they never run?  Should they be deleted, too?